dynparquet/reader.go

```diff
@@ -72,6 +73,63 @@ func (b *SerializedBuffer) DynamicRows() DynamicRowReader {
 	return Concat(b.fields, drg...).DynamicRows()
 }
 
+func (b *SerializedBuffer) String() string {
+	var buf bytes.Buffer
+	const (
+		minWidth = 15
+		tabWidth = 15
+		padding  = 2
+		padChar  = ' '
+		noFlags  = 0
+	)
+	w := tabwriter.NewWriter(&buf, minWidth, tabWidth, padding, padChar, noFlags)
+
+	truncateString := func(s string) string {
+		const ellipses = "..."
+		if len(s) > tabWidth {
+			return s[:tabWidth-len(ellipses)] + ellipses
+		}
+		return s
+	}
+
+	numRowGroups := b.NumRowGroups()
+	numRows := b.NumRows()
+	_, _ = w.Write([]byte(fmt.Sprintf("row groups: %d\ttotal rows: %d\n", numRowGroups, numRows)))
+	for i := 0; i < numRowGroups; i++ {
+		_, _ = w.Write([]byte("---\n"))
+		func() {
```

Comment:
Any specific reason for this `func()`? Is it for the `defer rows.Close()`?

Reply:
yes